### ISSUE TITLE
Error: Could not find navigation property ResourceGroup in entity type Resource when running generator for SAP Business One Service Layer Metadata

### ISSUE THREAD

**Issue by gregorwolf** [SAP/cloud-sdk#222]
**Describe the bug**

The first issue with the SAP Business One Service Layer Metadata was fixed in #176. But now there is another issue.

In the branch [collection-issue](https://github.com/gregorwolf/sap-business-one-odata-cap/tree/collection-issue) of my test project [sap-business-one-odata-cap](https://github.com/gregorwolf/sap-business-one-odata-cap) I've added "@sap-cloud-sdk/generator": "^1.22.0" and the script:

`generate-odata-client --inputDir srv/external --outputDir srv/odata-client --forceOverwrite` 

but when running this script I get the following error:

```
(node:12328) UnhandledPromiseRejectionWarning: Error: Could not find navigation property ResourceGroup in entity type Resource.
    at /Users/gwolf/Documents/Projects/cap/sap-business-one-odata-cap/node_modules/@sap-cloud-sdk/generator/dist/parser/v4/edmx-to-vdm.js:23:19
    at Array.map (<anonymous>)
    at navigationProperties (/Users/gwolf/Documents/Projects/cap/sap-business-one-odata-cap/node_modules/@sap-cloud-sdk/generator/dist/parser/v4/edmx-to-vdm.js:20:48)
    at /Users/gwolf/Documents/Projects/cap/sap-business-one-odata-cap/node_modules/@sap-cloud-sdk/generator/dist/parser/v4/edmx-to-vdm.js:32:200
    at Array.map (<anonymous>)
    at transformEntitiesV4 (/Users/gwolf/Documents/Projects/cap/sap-business-one-odata-cap/node_modules/@sap-cloud-sdk/generator/dist/parser/v4/edmx-to-vdm.js:32:29)
    at transformServiceMetadata (/Users/gwolf/Documents/Projects/cap/sap-business-one-odata-cap/node_modules/@sap-cloud-sdk/generator/dist/parser/service-parser.js:78:19)
    at parseService (/Users/gwolf/Documents/Projects/cap/sap-business-one-odata-cap/node_modules/@sap-cloud-sdk/generator/dist/parser/service-parser.js:45:12)
    at /Users/gwolf/Documents/Projects/cap/sap-business-one-odata-cap/node_modules/@sap-cloud-sdk/generator/dist/parser/service-parser.js:38:16
    at Array.map (<anonymous>)
(node:12328) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:12328) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**To Reproduce**
1.  checkout the branch [collection-issue](https://github.com/gregorwolf/sap-business-one-odata-cap/tree/collection-issue)
2. run `npm i`
3. run `npm run generate-odata-client`

**Expected behavior**
Also the SAP Business One Service Layer Metadata should be supported.

**Used Versions:**
 - node version v12.16.1
 - npm version 6.14.5
 - SAP Cloud SDK version 1.22.0

**Comment by marikaner** [comment]
Hey @gregorwolf,

thank you for trying the experimental version of our v4 generator. We currently do not yet support partner navigation properties, which might be related, but I don't think that this is the issue here.

The error above occurs when parsing this entity set:

```xml
<EntitySet EntityType="SAPB1.Resource" Name="Resources">
  <NavigationPropertyBinding Path="Items" Target="Items"/>
  <!-- Problematic line -->
  <NavigationPropertyBinding Path="ResourceGroup" Target="ResourceGroups"/>
  <NavigationPropertyBinding Path="Item" Target="Items"/>
  <NavigationPropertyBinding Path="ResourceCapacities" Target="ResourceCapacities"/>
</EntitySet>
```

This entity set references the following (shortened) entity type:
```xml
<EntityType Name="Resource">
  ...
  <NavigationProperty Name="Items" Partner="Resource" Type="Collection(SAPB1.Item)"/>
  <NavigationProperty Name="Item" Partner="Resources" Type="SAPB1.Item">
    <ReferentialConstraint Property="LinkedItem" ReferencedProperty="ItemCode"/>
  </NavigationProperty>
  <NavigationProperty Name="ResourceCapacities" Partner="Resource" Type="Collection(SAPB1.ResourceCapacity)"/>
</EntityType>
```

All of the navigation bindings from the entity set reference a navigation property in the entity type except for `ResourceGroup` which is causing your problem.
If you think this is correct, please point me to how this is actually linked in your service specification, if not, please fix this and try again.

**Comment by gregorwolf** [comment]
Dear @marikaner,

as I'm using the OData V4 API published for the for SAP Business One Service Layer at https://api.sap.com/api/Orders/overview I have to expect that this is correct as I can't change it. I will reach out to B1CloudAPISupport@sap.com to make them aware of this issue.

**Comment by marikaner** [comment]
Dear @gregorwolf,

when I download the , edmx from the api hub the "ResourceGroup" navigation property is there. Maybe it was updated in the meantime or you made some modifications?

**Comment by marikaner** [comment]
I tried generation though and it fails because of a different error. I will check this.

**Comment by marikaner** [comment]
@gregorwolf We fixed this different error and I made sure that generation for this service works. Let us know if you encounter any other issues.

**Comment by gregorwolf** [comment]
Hi @marikaner,

I've updated the master branch of my sample project to "@sap-cloud-sdk/generator": "^1.24.0". Unfortunately I can't replicate your successful generation. I get the following error:

[2020-07-07T18:54:23.720Z] INFO     (generator): Generating complex type: RclRecurringExecutionParams...
(node:13655) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'leadingTrivia' of undefined
    at NamedImportExportSpecifierStructurePrinter.printLeadingTrivia (/home/gwolf/Dokumente/Projects/sapcp/sap-business-one-odata-cap/node_modules/ts-morph/dist/ts-morph.js:6095:40)
    at NamedImportExportSpecifierStructurePrinter.printText (/home/gwolf/Dokumente/Projects/sapcp/sap-business-one-odata-cap/node_modules/ts-morph/dist/ts-morph.js:6089:14)
    at printTextWithSeparator (/home/gwolf/Dokumente/Projects/sapcp/sap-business-one-odata-cap/node_modules/ts-morph/dist/ts-morph.js:6157:21)
    at CommaSeparatedStructuresPrinter.printText (/home/gwolf/Dokumente/Projects/sapcp/sap-business-one-odata-cap/node_modules/ts-morph/dist/ts-morph.js:6142:9)
    at NamedImportExportSpecifierStructurePrinter.printTexts (/home/gwolf/Dokumente/Projects/sapcp/sap-business-one-odata-cap/node_modules/ts-morph/dist/ts-morph.js:8152:33)
    at NamedImportExportSpecifierStructurePrinter.printTextsWithBraces (/home/gwolf/Dokumente/Projects/sapcp/sap-business-one-odata-cap/node_modules/ts-morph/dist/ts-morph.js:8139:14)
    at ImportDeclarationStructurePrinter.printTextInternal (/home/gwolf/Dokumente/Projects/sapcp/sap-business-one-odata-cap/node_modules/ts-morph/dist/ts-morph.js:8083:58)
    at ImportDeclarationStructurePrinter.printText (/home/gwolf/Dokumente/Projects/sapcp/sap-business-one-odata-cap/node_modules/ts-morph/dist/ts-morph.js:6091:14)
    at StatementStructurePrinter.printText (/home/gwolf/Dokumente/Projects/sapcp/sap-business-one-odata-cap/node_modules/ts-morph/dist/ts-morph.js:8241:53)
    at StatementStructurePrinter.printTexts (/home/gwolf/Dokumente/Projects/sapcp/sap-business-one-odata-cap/node_modules/ts-morph/dist/ts-morph.js:8211:22)
(node:13655) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:13655) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

Looking forward for further tips.

Best regards
Gregor

**Comment by marikaner** [comment]
Oh, I mistakenly used a different edmx file to try this. I was able to reproduce your problem and will try to fix it.

**Comment by marikaner** [comment]
@gregorwolf The issue has been fixed in version 1.24.1. Please note, that we do not yet support enum types.

**Comment by gregorwolf** [comment]
Thank you @marikaner I've where able to create a first client [sapb-1-service](https://github.com/gregorwolf/sap-business-one-odata-cap/tree/master/srv/odata-client/sapb-1-service)
